operator/src/main/java/dev/responsive/k8s/crd/DemoPolicy.java

```diff
@@ -0,0 +1,133 @@
+package dev.responsive.k8s.crd;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class DemoPolicy {
+  public static class ProcessingRateDiagnoser {
+    private final int rate;
+    private final OptionalInt windowMs;
+
+    public int getRate() {
+      return rate;
+    }
+
+    public OptionalInt getWindowMs() {
+      return windowMs;
+    }
+
+    public ProcessingRateDiagnoser(
+        @JsonProperty("rate") final int rate,
+        @JsonProperty("windowMs") final OptionalInt windowMs
+    ) {
+      this.rate = rate;
+      this.windowMs = Objects.requireNonNull(windowMs);
+    }
+  }
+
+  public static class Diagnoser {
+    public enum Type {
+      PROCESSING_RATE_SCALE_UP,
+      PROCESSING_RATE_SCALE_DOWN,
+      LAG_SCALE_UP
+    }
+
+    private final Type type;
+    private final Optional<ProcessingRateDiagnoser> processingRateScaleUp;
+    private final Optional<ProcessingRateDiagnoser> processingRateScaleDown;
+
+    @JsonCreator
+    public Diagnoser(
```

Comment:
I think this should probably use `JsonSubTypes` and allowing us to multiplex instead of having each diagnoser be passed in as optional. https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/KsqlEntity.java is a good example of this

Reply:
I tried but I couldn't get the yaml serialization/deserialization to work =/

Reply:
I think the problem is that the schema generator, which generates the CRD yaml schema doesn't know how to handle polymorphism. Kind of a bummer.